cpp/src/arrow/csv/writer_test.cc

```diff
@@ -37,24 +38,34 @@ struct WriterTestParams {
   std::shared_ptr<Schema> schema;
   std::string batch_data;
   WriteOptions options;
-  bool options_invalid;
+  // When set, expect the test to fail just from the supplied WriteOptions alone.
+  bool expect_invalid_options;
   std::string expected_output;
+  // When set, expect the test to fail with this string.
+  util::optional<std::string> expect_invalid_string;
```

Comment:
can you use a status memory instead of the two separate additional parameter here?